### PR TITLE
Add mkldnn softmax operator

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

File: aten/src/ATen/native/mkldnn/SoftMax.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+
+namespace at {
+namespace native {
+
+namespace {
+// TODO: move this to ideep
```
Comment (bddppq) [inline aten/src/ATen/native/mkldnn/SoftMax.cpp]: @Jianhui-Li @XiaobingSuper @gujinghui

**Review by gchanan — APPROVED** [review]
would be good to have an MKLDNN person look at those details.

**Comment by bddppq** [comment]
@Jianhui-Li @XiaobingSuper @gujinghui Mind taking a look at the `ideep_softmax_forward`? ideally this will be moved to ideep itself.

**Comment by gujinghui** [comment]
@bddppq 
Sure. Moved to ideep in PR #20569. Pls help review.

**Pull request closed without merging**